ide/xml.xam/src/org/netbeans/modules/xml/xam/AbstractModel.java

```diff
@@ -179,12 +180,22 @@ public State getState() {
         return status;
     }
     
+    @Override
+    public String getStatusMessage() {
+        return statusMessage;
+    }
+    
     protected void setState(State s) {
+        setState(s, null);
+    }
+
+    protected void setState(State s, String message) {
         if (s == status) {
```

Comment:
`statusMessage` does not behave as a property; can be only updated during a state change ... which can be reasonable. Pls. document the behaviour, `status` name looks like property updatable even during a single state.